pentaho-js/src/main/javascript/cdf/components/VisualizationAPIComponent.js

```diff
@@ -25,24 +30,49 @@ define([
 
     render: function(data) {
       var domElem = this.placeholder()[0];
-      var wrapper = new VisualWrapper(domElem);
-      wrapper.data = data;
-      wrapper.visualSpec = this.getVisualSpec();
-      wrapper.update()
-        .then(_.bind(this.endExec, this), _.bind(this.failExec, this));
-    },
-  	
-    getVisualSpec: function() {
-      var visualSpec = {};
-      
-      _.each(this.vizOptions, function(option) {
-        visualSpec[option[0]] = this.getParameterValue(option[1]);
-      }, this.dashboard);
+      var contextVars = new GlobalContextVars({application: "pentaho-cdf"});
+
+      var _context = new Context(contextVars);
+      var me = this;
+      _context.getAsync(this.vizId)
+          .then( function (Model) {
+            var createdModel = new Model({});
+            createdModel.data = new Table(data);
+            createdModel.height= me.height;
+            createdModel.width = me.width;
+
+            $.each(me.vizOptions, function(i, v) {
+              var prop = createdModel.get(v[0]);
+              var value = me.dashboard.getParameterValue(v[1]);
 
-      visualSpec.type = this.vizId;
+              if (prop.type.isSimple) {
+                createdModel[prop] = value;
```

Comment:
Please do not use the JavaScript property getter/setter directly with generic code, cause these aren't defined when the name of the property is already taken in the class, for some other use. The generic safe way is to use the set (and get, for getting) method:
```js
createdModel.set(v[1], value);
```